core/src/main/scala/cats/data/ReaderWriterStateT.scala

```diff
@@ -0,0 +1,554 @@
+    transform { (l, s, a) => (l, s, f(a)) }
+
+  /**
+   * Modify the written log value using `f`.
+   */
+  def mapWritten[LL](f: L => LL)(implicit F: Functor[F]): ReaderWriterStateT[F, E, S, LL, A] =
+    transform { (l, s, a) => (f(l), s, a) }
+
+  /**
+   * Combine this computation with `rwsb` using `fn`. The state will be be threaded
+   * through the computations and the log values will be combined.
+   */
+  def map2[B, Z](rwsb: ReaderWriterStateT[F, E, S, L, B])(fn: (A, B) => Z)(
+    implicit F: FlatMap[F], L: Semigroup[L]): ReaderWriterStateT[F, E, S, L, Z] =
+    flatMap { a =>
+      rwsb.map { b =>
+        fn(a, b)
+      }
+    }
+
+  /**
+   * Modify the result of the computation by feeding it into `f`, threading the state
+   * through the resulting computation and combining the log values.
+   */
+  def flatMap[B](f: A => ReaderWriterStateT[F, E, S, L, B])(
+    implicit F: FlatMap[F], L: Semigroup[L]): ReaderWriterStateT[F, E, S, L, B] =
+    ReaderWriterStateT.applyF {
+      F.map(runF) { rwsfa =>
+        (e: E, s0: S) =>
+          F.flatMap(rwsfa(e, s0)) { case (la, sa, a) =>
+            F.flatMap(f(a).runF) { rwsfb =>
+              F.map(rwsfb(e, sa)) { case (lb, sb, b) =>
+                (L.combine(la, lb), sb, b)
+              }
+            }
+          }
+      }
+    }
+
+  /**
+   * Like [[map]], but allows the mapping function to return an effectful value.
+   */
+  def flatMapF[B](faf: A => F[B])(implicit F: FlatMap[F]): ReaderWriterStateT[F, E, S, L, B] =
+    ReaderWriterStateT.applyF {
+      F.map(runF) { rwsfa =>
+        (e: E, s: S) =>
+          F.flatMap(rwsfa(e, s)) { case (l, s, a) =>
+            F.map(faf(a))((l, s, _))
+          }
+      }
+    }
+
+  /**
+   * Transform the resulting log, state and value using `f`.
+   */
+  def transform[LL, B](f: (L, S, A) => (LL, S, B))(implicit F: Functor[F]): ReaderWriterStateT[F, E, S, LL, B] =
+    ReaderWriterStateT.applyF {
+      F.map(runF) { rwsfa =>
+        (e: E, s: S) => F.map(rwsfa(e, s)) { case (l, s, a) =>
+          val (ll, sb, b) = f(l, s, a)
+          (ll, sb, b)
+        }
+      }
+    }
+
+  /**
+   * Like [[transform]], but allows the context to change from `F` to `G`.
+   */
+  def transformF[G[_], LL, B](f: F[(L, S, A)] => G[(LL, S, B)])(
+    implicit F: Monad[F], G: Applicative[G]): ReaderWriterStateT[G, E, S, LL, B] =
+    ReaderWriterStateT.apply((e, s) => f(run(e, s)))
+
+
+  /**
+   * Modify the resulting state.
+   */
+  def modify(f: S => S)(implicit F: Functor[F]): ReaderWriterStateT[F, E, S, L, A] =
+    transform { (l, s, a) => (l, f(s), a) }
+
+  /**
+   * Inspect a value from the input state, without modifying the state.
+   */
+  def inspect[B](f: S => B)(implicit F: Functor[F]): ReaderWriterStateT[F, E, S, L, B] =
+    transform { (l, s, a) => (l, s, f(s)) }
+
+  /**
+   * Get the input state, without modifying it.
+   */
+  def get(implicit F: Functor[F]): ReaderWriterStateT[F, E, S, L, S] =
```

Comment:
The corresponding method in `State` is named `get`, but for RWST I wonder if `getS` (or perhaps `getState`) might be a better name. In particular, this will help distinguish between getting the input `E` and getting the input `S`.